clients/segments/segment_test.go

```diff
@@ -342,13 +310,47 @@ func TestUpsert(t *testing.T) {
 			}, nil)
 
 		fsClient := segments.NewTestClient(mockClient)
-		resp, err := fsClient.Upsert(ctx, "", []byte(payload))
+		resp, err := fsClient.Create(ctx, []byte(payload))
 
 		assert.NotEmpty(t, resp)
 		assert.NoError(t, err)
 		assert.Equal(t, apiResponse, string(resp.Data))
 		assert.Equal(t, http.StatusCreated, resp.StatusCode)
 	})
+}
+
+func TestUpdate(t *testing.T) {
```

Comment:
Because of editing the given payload, I would like to have a test which tests (and documents) what has the precedence, given or get value for `owner` and `version`.